### PR TITLE
feat: Add support for rendering dynamic imports

### PR DESCRIPTION
**Pull request by mattsjones**
- Render all dynamic imports through [loadable-components](https://github.com/smooth-code/loadable-components)
- All scripts are now loaded async for concurrent downloads
- Render`link rel="preload"` tags for all tags
- Above features allow efficient braid theming across multiple brands

Note: most of this PR is test updates

BREAKING CHANGE: Render and Server API now need to use the new SkuProvider component for correct script/style rendering.

### CODE REVIEW

File: config/babel/babelConfig.js [hunk 1]
```diff
@@ -28,13 +28,12 @@ module.exports = ({ target, lang = 'js' }) => {
...
     ],
-    require.resolve('babel-plugin-macros')
+    require.resolve('babel-plugin-macros'),
+    require.resolve('@loadable/babel-plugin')
   ];
 
   if (isBrowser) {
     plugins.push(require.resolve('babel-plugin-seek-style-guide'));
-  } else {
-    plugins.push(require.resolve('babel-plugin-dynamic-import-node'));
```
Comment (markdalgleish) [inline config/babel/babelConfig.js]: 👋👋👋

File: context/defaultClientEntry.js [hunk 2]
```diff
@@ -0,0 +1 @@
+module.exports = 'main';
```
Comment (markdalgleish) [inline context/defaultClientEntry.js]: Why do we need this? Feel like this could use a comment.

File: entry/render/buildRenderEntry.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable import/no-unresolved */
```
Comment (markdalgleish) [inline entry/render/buildRenderEntry.js]: This is operating on the whole file. Should we scope it to the `'__sku_alias__renderEntry'` import?

File: context/defaultClientEntry.js (hunk 2 above)
Comment (mattsjones) [reply context/defaultClientEntry.js]: Well it's shared in 3 places now. Felt like it should be centrally defined.

File: test/test-cases/custom-src-paths/__snapshots__/custom-src-paths.test.js.snap [hunk 4]
```diff
@@ -6,13 +6,17 @@ Object {
     <meta charset=\\"UTF-8\\">
     <title>hello-world</title>
     <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
-    
+    <link data-chunk=\\"main\\" rel=\\"preload\\" as=\\"script\\" href=\\"http://localhost:4002/runtime-27f87d9f0d19bf7deb87.js\\">
```
Comment (markdalgleish) [inline test/test-cases/custom-src-paths/__snapshots__/custom-src-paths.test.js.snap]: This is so cool 😍

Comment (markdalgleish) [reply test/test-cases/custom-src-paths/__snapshots__/custom-src-paths.test.js.snap]: We're losing `crossorigin="anonymous"`—is this a problem?

Comment (markdalgleish) [reply test/test-cases/custom-src-paths/__snapshots__/custom-src-paths.test.js.snap]: ☝️ Oops, commented on the wrong line—but the point still stands.

Comment (mattsjones) [reply test/test-cases/custom-src-paths/__snapshots__/custom-src-paths.test.js.snap]: Working on that atm. That's what the loadable-component PR is for

File: test/test-cases/typescript-css-modules/typescript-css-modules.test.js [hunk 5]
```diff
@@ -65,7 +65,7 @@ describe('typescript-css-modules', () => {
     });
 
     it('should generate the expected files', async () => {
-      const files = await dirContentsToObject(distDir);
+      const files = await dirContentsToObject(distDir, ['.js', '.css']);
```
Comment (markdalgleish) [inline test/test-cases/typescript-css-modules/typescript-css-modules.test.js]: Why is this being limited to JS and CSS now?

File: test/test-cases/typescript-css-modules/app/src/server.tsx [hunk 6]
```diff
@@ -4,23 +4,34 @@ import { renderToString } from 'react-dom/server';
...
+
     res.send(`
       <!DOCTYPE html>
       <html>
         <head>
           <meta charset="UTF-8">
           <title>My Awesome Project</title>
           <meta name="viewport" content="width=device-width, initial-scale=1">
-          ${headTags}
+          ${getHeadTags()}
```
Comment (markdalgleish) [inline test/test-cases/typescript-css-modules/app/src/server.tsx]: Why is this now a function call?

File: test/test-cases/zero-config/zero-config.test.js [hunk 7]
```diff
@@ -1,50 +0,0 @@
...
-    });
-  });
-
-  describe('lint', () => {
-    it('should lint successfully', async () => {
-      const { childProcess } = await runSkuScriptInDir('lint', __dirname);
-      expect(childProcess.exitCode).toEqual(0);
-    });
-  });
-});
```
Comment (markdalgleish) [inline test/test-cases/zero-config/zero-config.test.js]: 👋👋👋

File: test/test-cases/typescript-css-modules/typescript-css-modules.test.js (hunk 5 above)
Comment (mattsjones) [reply test/test-cases/typescript-css-modules/typescript-css-modules.test.js]: We now spit out a `webpackStats.json` file which contains info that will be specific to the machine that created it so it's not a good candidate for snapshots. The function currently only has a whitelist so I just whitelisted everything else.

File: test/test-cases/typescript-css-modules/app/src/server.tsx (hunk 6 above)
Comment (mattsjones) [reply test/test-cases/typescript-css-modules/app/src/server.tsx]: Previously we created a list of assets for a build. The assets are now specific to each render so a function is required.

File: test/test-cases/typescript-css-modules/typescript-css-modules.test.js (hunk 5 above)
Comment (markdalgleish) [reply test/test-cases/typescript-css-modules/typescript-css-modules.test.js]: Do we only output this file for server rendering?

Comment (mattsjones) [reply test/test-cases/typescript-css-modules/typescript-css-modules.test.js]: Yes

Comment (markdalgleish) [reply test/test-cases/typescript-css-modules/typescript-css-modules.test.js]: 😎

**Pull request merged**